migrations/versions/a7a9acdc0670_make_keys_unique.py

```diff
@@ -0,0 +1,16 @@
+"""Make meme template keys unique."""
+
+from alembic import op
+import sqlalchemy as sa
+
+
+revision = 'a7a9acdc0670'
+down_revision = '7da56a4eb387'
+
+
+def upgrade():
+    op.create_unique_constraint(None, 'memes', ['key'])
+
+
+def downgrade():
+    op.drop_constraint(None, 'memes', type_='unique')
```

Comment:
You will need to specify the name of the constraint if you want the `downgrade` operation to work. Alembic will auto-assign a name if you use `None` when creating the constraint, but it can't determine the name accurately enough for the downgrade. In this case, it called it `memes_key_key`